### PR TITLE
Fix rendering of notification badge in expanded dropdown

### PR DESCRIPTION
**Pull request by tjmw**
## What does this change?

Fixes the styling of a dropdown item with a notification.

## Why?

The styles were changed in #7671 to increase the clickable area of each menu item. But the removal of a `display: flex` had the effect of causing the badge for a menu item (when expanded) to appear below the item instead of to the right as intended. This fixes the issue, while keeping the larger clickable area.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |
| ![beforemobile][] | ![aftermobile][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/379839/34385aca-f9a0-48b6-93e3-017be2868bba
[after]: https://github.com/guardian/dotcom-rendering/assets/379839/8d893f9a-3a52-4d45-ab62-892e162aa2ab
[beforemobile]: https://github.com/guardian/dotcom-rendering/assets/379839/b96e8d59-ac8e-47e9-a1bf-1a6cbfac8f7b
[aftermobile]: https://github.com/guardian/dotcom-rendering/assets/379839/b1e28df4-06bd-40c3-8894-c33d3adb0883

(See #6556 for the original intended design).

### CODE REVIEW

File: dotcom-rendering/src/components/Dropdown.tsx [hunk 1]
```diff
@@ -35,6 +35,7 @@ interface Props {
 	dataLinkName: string;
 	cssOverrides?: SerializedStyles;
 	children?: React.ReactNode;
+	defaultIsExpanded?: boolean;
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Dropdown.tsx:38]: Is this done in order to simulate user interaction?

I’d recommend using `@testing-library/react`, as we do in [`AustralianTerritorySwitcher`](https://github.com/guardian/dotcom-rendering/blob/2ed9dd5f731246a87274421dece72961341fa96f/dotcom-rendering/src/components/AustralianTerritorySwitcher.importable.stories.tsx#L17-L24) and [`Nav`](https://github.com/guardian/dotcom-rendering/blob/2ed9dd5f731246a87274421dece72961341fa96f/dotcom-rendering/src/components/Nav/Nav.stories.tsx#L123-L133)

File: dotcom-rendering/src/components/Dropdown.tsx [hunk 2]
```diff
@@ -208,6 +209,14 @@ const dropdownButtonNotificationBadgeStyles = css`
 	margin-top: -3px;
 `;
 
+const dropdownItemContainerStyles = css`
+	display: flex;
+`;
+
+const dropdownItemTextContainerStyles = css`
+	display: block;
+`;
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Dropdown.tsx:218]: Would moving the `NotificationBadge` inside the anchor tag not be sufficient?

File: dotcom-rendering/src/components/Dropdown.tsx (hunk 1 above)
Comment (tjmw) [reply dotcom-rendering/src/components/Dropdown.tsx:38]: Ooh interesting, can I use that in the context of a story? I'm particularly interested in having a Chromatic snapshot for this component in the expanded state as there's been a visual regression here (which I think Chromatic would have picked up).

File: dotcom-rendering/src/components/Dropdown.tsx (hunk 2 above)
Comment (tjmw) [reply dotcom-rendering/src/components/Dropdown.tsx:218]: I think I tried that but it didn't have the effect I needed.

File: dotcom-rendering/src/components/Dropdown.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/components/Dropdown.tsx:38]: Yes, it’s a pretty useful feature: https://storybook.js.org/docs/writing-tests/interaction-testing

Comment (tjmw) [reply dotcom-rendering/src/components/Dropdown.tsx:38]: 👍 Nice, thanks. I'll investigate.

File: dotcom-rendering/src/components/Dropdown.tsx [hunk 3]
```diff
@@ -362,25 +367,21 @@ const DropdownLink = ({ link, index }: DropdownLinkProps) => {
...
+					{link.notifications?.map((notification) => (
+						<NotificationMessage
+							notification={notification}
+							key={notification.id}
+						/>
+					))}
 				</div>
-			)}
+				{hasNotifications && (
+					<div style={{ display: 'block' }}>
```
Comment (mxdvl) [inline dotcom-rendering/src/components/Dropdown.tsx:380]: I think it would be preferable to set `flex-shrink: 0` on the `NotificationBadge`

**Review by mxdvl — APPROVED** [review]
No blockers left — we can address the testing library later

File: dotcom-rendering/src/components/Dropdown.tsx (hunk 3 above)
Comment (tjmw) [reply dotcom-rendering/src/components/Dropdown.tsx:380]: Oh cool, I didn't know about `flex-shrink` 👍

Comment (tjmw) [reply dotcom-rendering/src/components/Dropdown.tsx:380]: That worked well, thanks for the suggestion.

**Comment by tjmw** [comment]
> No blockers left — we can address the testing library later

Thanks @mxdvl. I'll merge and deploy this morning and tackle the story/testing library work in a follow up PR.

**Pull request merged**